the_kybra_book/src/http.md

```diff
@@ -93,24 +110,28 @@ def eth_get_balance(ethereum_address: str) -> Async[JSON]:
     cycle_cost_per_byte = 300_000  # TODO not sure on this exact cost
     cycle_cost_total = cycle_cost_base + cycle_cost_per_byte * max_response_bytes
 
-    http_result: CallResult[HttpResponse] = yield management_canister.http_request({
-        'url': stable_storage.get('ethereum_url') or '',
-        'max_response_bytes': max_response_bytes,
-        'method': {'post': None},
-        'headers': [],
-        'body': f'{{"jsonrpc":"2.0","method":"eth_getBalance","params":["{ethereum_address}","earliest"],"id":1}}'.encode('utf-8'),
-        'transform': {
-            'function': (ic.id(), 'eth_transform'),
-            'context': bytes()
-        },
-    }).with_cycles(cycle_cost_total)
-
-    if http_result.err is not None:
-        if http_result.err:
-            ic.trap(http_result.err)
-        ic.trap('http_result had an error')
-
-    return http_result.ok['body'].decode('utf-8')
+    http_result: CallResult[HttpResponse] = yield management_canister.http_request(
+        {
+            "url": stable_storage.get("ethereum_url") or "",
+            "max_response_bytes": max_response_bytes,
+            "method": {"post": None},
+            "headers": [],
+            "body": f'{{"jsonrpc":"2.0","method":"eth_getBalance","params":["{ethereum_address}","earliest"],"id":1}}'.encode(
+                "utf-8"
+            ),
+            "transform": {"function": (ic.id(), "eth_transform"), "context": bytes()},
+        }
+    ).with_cycles(cycle_cost_total)
+
+    def handle_http_result_err(err: str):
```

Comment:
We don't need the function here, we can just do the ic.trap(err)

Reply:
I need to talk to Jordan about this one

Reply:
I talked to Jordan and he informed me that the old nested if was only to make Pylance happy, now that we have the match we don't need it and we can simplify to just trapping immediately with the error.